### PR TITLE
articles: add edge cases for slice to array pointer conversion

### PR DESCRIPTION
**Pull request by cuonglm**
See: https://github.com/golang/go/blob/052da5717e02659da49707873b3868fe36f2aaf0/doc/go_spec.html#L4321

### CODE REVIEW

File: articles/container.html [hunk 1]
```diff
@@ -1491,19 +1491,24 @@ <h3>Convert Slice to Array Pointer</h3>
...
 
 func main() {
-    var x = make([]int, 3, 5)
-    var _ = (*[3]int)(x) // okay
-    var _ = (*[2]int)(x) // okay
-    var _ = (*A)(x)      // okay
-    var _ = P(x)         // okay
+    var x []int
+    var y = make([]int, 0)
+    var x0 = (*[0]int)(x) // okday, x0 == nil
```
Comment (go101) [inline articles/container.html:1496]: "okday", a typo, ;D

And it looks a line "_, _ = x0, y0" is need here.

File: articles/container.html [hunk 2]
```diff
@@ -1491,19 +1491,24 @@ <h3>Convert Slice to Array Pointer</h3>
...
-    var _ = (*A)(y)      // okay
-    var _ = P(y)         // okay
+    var z = make([]int, 3, 5)
+    var _ = (*[3]int)(z) // okay
+    var _ = (*[2]int)(z) // okay
+    var _ = (*A)(z)      // okay
+    var _ = P(z)         // okay
     
-    var _ = (*[4]int)(x) // will panic
+    var t = S(x)
```
Comment (go101) [inline articles/container.html:1505]: S(z)? Othewise, following lines will panic.

BTW, how about to use w instead of t?

**Review by go101 — CHANGES_REQUESTED** [review]
@cuonglm
Thanks very much for the improvement!

It looks there are two places still needing tweaks.

File: articles/container.html (hunk 1 above)
Comment (cuonglm) [reply articles/container.html]: Done.

File: articles/container.html (hunk 2 above)
Comment (cuonglm) [reply articles/container.html]: Yeah, `w` is better. Done.

**Review by go101 — APPROVED** [review]
@cuonglm
Many thanks for the improvement.

**Pull request merged**